pkg/controller/llmisvc/workload_tls_self_signed.go

```diff
@@ -162,6 +277,86 @@ func isCertificateExpired(curr *corev1.Secret) bool {
 	return false
 }
 
+func isSameSANs(curr *corev1.Secret, expected string) bool {
+	sans, ok := curr.Annotations[certificatesSANsAnnotation]
+	if ok {
+		return sans == expected
+	}
+	return false
+}
+
+func isCertificateCorrupted(curr *corev1.Secret) bool {
+	return len(curr.Data["tls.key"]) == 0 || len(curr.Data["tls.crt"]) == 0
+}
+
+func (r *LLMInferenceServiceReconciler) collectDNSNames(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService) []string {
+	dnsNames := []string{
+		"localhost", // P/D sidecar sends requests for decode over localhost
+		network.GetServiceHostname(kmeta.ChildName(llmSvc.GetName(), "-kserve-workload-svc"), llmSvc.GetNamespace()),
+		fmt.Sprintf("%s.%s.svc", kmeta.ChildName(llmSvc.GetName(), "-kserve-workload-svc"), llmSvc.GetNamespace()),
+	}
+
+	if llmSvc.Spec.Router != nil && llmSvc.Spec.Router.Scheduler != nil && llmSvc.Spec.Router.Scheduler.Pool != nil {
+		ipSpec := llmSvc.Spec.Router.Scheduler.Pool.Spec
```

Comment:
nit, but maybe worth expanding to `infPool` at least? I got it confused with IP address at first :)